contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -1945,17 +1951,29 @@ select_common_type_setop(ParseState *pstate, List *exprs, Node **which_expr, con
 	Node		*result_expr = (Node*) linitial(exprs);
 	Oid			result_type = InvalidOid;
 	ListCell	*lc;
-        bool             is_case_expr = (strlen(context) == 4 && strncmp(context, "CASE", 4) == 0);
+	bool		is_case_expr = (strlen(context) == 4 && strncmp(context, "CASE", 4) == 0);
 
 	/* Find a common type based on precedence. NULLs are ignored, and make 
 	 * string literals varchars. If a type besides CHAR, NCHAR, VARCHAR, 
 	 * or NVARCHAR is present, let engine handle finding the type.
 	 * But if it is CASE expr then it will also check for text and ntext.
-         */
+	 */
 	foreach(lc, exprs)
 	{
-		Node	*expr = (Node *) lfirst(lc);
+		Node		*expr = (Node *) lfirst(lc);
 		Oid		type = exprType(expr);
+		if (is_case_expr)
+		{
+			Oid		baseType = get_immediate_base_type_of_UDT_internal(type);
+			/*
+			 * If any of the branch is of UDT or SYSNAME, then we will find the baseType using
```

Comment:
Lets add separate comment for UDT and SYSNAME, for UDT we will consider its base oid using `get_immediate_base_type_of_UDT_internal` and for sysname we will use varchar.